lib/codecs/src/encoding/mod.rs

```diff
@@ -241,6 +241,32 @@ pub enum Serializer {
     RawMessage(RawMessageSerializer),
 }
 
+impl Serializer {
+    /// Check if the serializer supports encoding to JSON via `Serializer::encode_json`.
+    pub fn supports_json(&self) -> bool {
+        match self {
+            Serializer::Json(_) | Serializer::NativeJson(_) => true,
+            Serializer::Native(_) | Serializer::RawMessage(_) => false,
+        }
+    }
+
+    /// Encode event as JSON.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the serializer does not support encoding to JSON. Call `Serializer::supports_json`
+    /// if you need to determine the capability to encode to JSON at runtime.
+    pub fn encode_json(&self, event: Event) -> Result<serde_json::Value, serde_json::Error> {
```

Comment:
This name is somewhat confusing. I might call it something like `to_json_value` or similar to differentiate it from normal encoding.

Reply:
Done with c0bcf7d3ca5e1c54c575a39f6494ffd05feac0dc